From 4bfdd677e3e398651841a4bd2d5b01124d442ffd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sat, 14 Mar 2020 14:45:42 +0100
Subject: [PATCH] calendar: Adjust for evolution changes

When launching the default calendar application, we special-case
evolution to make sure it starts up with the calendar component.

This is currently broken in two ways:

 - evolution changed its .desktop file to use reverse DNS notation

 - as evolution can now be distributed via flatpak, we can no longer
   assume that 'evolution-calendar.desktop' exists when evolution does
   (even though we ship the .desktop file ourselves, it is considered
   invalid if the executable isn't found)

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1093
---
 js/ui/calendar.js | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/js/ui/calendar.js b/js/ui/calendar.js
index 524d2dd32d..aa1c4558a5 100644
--- a/js/ui/calendar.js
+++ b/js/ui/calendar.js
@@ -931,10 +931,13 @@ class EventsSection extends MessageList.MessageListSection {
         Main.overview.hide();
         Main.panel.closeCalendar();
 
-        let app = this._getCalendarApp();
-        if (app.get_id() == 'evolution.desktop')
-            app = Gio.DesktopAppInfo.new('evolution-calendar.desktop');
-        app.launch([], global.create_app_launch_context(0, -1));
+        let appInfo = this._getCalendarApp();
+        if (appInfo.get_id() === 'org.gnome.Evolution.desktop') {
+            let app = Gio.DesktopAppInfo.new('evolution-calendar.desktop');
+            if (app)
+                appInfo = app;
+        }
+        appInfo.launch([], global.create_app_launch_context(0, -1));
     }
 
     setDate(date) {
-- 
2.24.1

